test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/LinqIntegrationTests.cs

```diff
@@ -164,6 +171,24 @@ internal class DerivedEntityWithTwoKeyProperties : EntityWithTwoKeyProperties
 
         [Fact]
         public void TwoWhereClausesForKeysShouldReturnUrlForSingleton()
+        {
+            // Simple case, only two key properties provided
+            string query1 = context.CreateQuery<EntityWithTwoDiffOrderKeyProperties>("Test")
+                .Where(p => p.ID1 == 1)
+                .Where(p => p.ID2 == 2)
+                .ToString();
+
+            string query2 = context.CreateQuery<EntityWithTwoDiffOrderKeyProperties>("Test")
+                .Where(p => p.ID2 == 2)
+                .Where(p => p.ID1 == 1)
+                .ToString();
+
+            Assert.Equal(query1, query2);
+            Assert.Equal(rootUrl + "Test(ID2=2,ID1=1)", query1);

```

Comment:
A little confusing.
Internally what dictates the order of the keys?

Reply:
I believe it's the `Key` attribute on the corresponding entity class:

```c#
[Key("ID2", "ID1")]
```

Reply:
On the CSDL, it's the order where the key property is present.
On the C# class from the Code generator, it's the KeyAttribute string (As @habbes mentioned).